### PR TITLE
Feature: (API) Retrieve BGP sessions by ID

### PR DESCRIPTION
**Pull request by Lee303**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

This PR adds an API endpoint for retrieving BGP sessions by ID.

### CODE REVIEW

File: html/includes/api_functions.inc.php [hunk 1]
```diff
@@ -506,6 +506,31 @@ function list_bgp()
...
+
+    $app        = \Slim\Slim::getInstance();
+    $router = $app->router()->getCurrentRoute()->getParams();
+
+    $bgpPeerId        = $router['id'];
+    if (!is_numeric($bgpPeerId)) {
+        api_error(400, 'Invalid id has been provided');
+    }
+
+    $bgp_session       = dbFetchRows("SELECT `bgpPeers`.* FROM `bgpPeers` LEFT JOIN `devices` ON `bgpPeers`.`device_id` = `devices`.`device_id` WHERE `bgpPeerState` IS NOT NULL AND `bgpPeerState` != '' AND bgpPeer_id = ?", array($bgpPeerId));
```
Comment (murrant) [inline html/includes/api_functions.inc.php]: Why are you joining the devices table here if you only return data from the bgpPeers table?

File: html/includes/api_functions.inc.php [hunk 2]
```diff
@@ -506,6 +506,31 @@ function list_bgp()
...
 
 
+function get_bgp()
+{
+    check_is_read();
+
+    $app        = \Slim\Slim::getInstance();
+    $router = $app->router()->getCurrentRoute()->getParams();
+
+    $bgpPeerId        = $router['id'];
```
Comment (murrant) [inline html/includes/api_functions.inc.php]: How do you figure out the bgpPeer_id via the API?

What if I want to look up by bgpPeerIdentifier ?

File: html/includes/api_functions.inc.php (hunk 1 above)
Comment (Lee303) [reply html/includes/api_functions.inc.php]: My bad, this was copied and adapted from the list_bgp() function. Will fix shortly.

File: html/includes/api_functions.inc.php (hunk 2 above)
Comment (Lee303) [reply html/includes/api_functions.inc.php]: This is returned from GET /api/v0/bgp (list_bgp()), and I chose this for get_bgp() as this is the primary key and uniquely identifies the BGP session. I believe bgpPeerIdentifier would be best suited to to an additional filter for list_bgp()? Apologies if I've missed the mark here, this is my first venturing into librenms :)

Comment (laf) [reply html/includes/api_functions.inc.php]: > What if I want to look up by bgpPeerIdentifier ?

That's not a unique reference within the DB.

**Comment by laf** [comment]
Thanks for doing this spotts, I'll give this a quick test later :)

File: html/includes/api_functions.inc.php (hunk 1 above)
Comment (laf) [reply html/includes/api_functions.inc.php]: Fixed for you :)

**Pull request merged**